src/sources/opentelemetry/grpc.rs

```diff
@@ -2,17 +2,26 @@ use futures::TryFutureExt;
 use opentelemetry_proto::proto::collector::logs::v1::{
     logs_service_server::LogsService, ExportLogsServiceRequest, ExportLogsServiceResponse,
 };
-use vector_core::config::LogNamespace;
+use tonic::{Request, Response, Status};
+use vector_common::internal_event::{CountByteSize, InternalEventHandle as _, Registered};
 use vector_core::{
+    config::LogNamespace,
     event::{BatchNotifier, BatchStatus, BatchStatusReceiver, Event},
     EstimatedJsonEncodedSizeOf,
 };
 
-#[derive(Debug, Clone)]
-pub(crate) struct Service {
-    pub(crate) pipeline: SourceSender,
-    pub(crate) acknowledgements: bool,
-    pub(crate) log_namespace: LogNamespace,
+use crate::{
+    internal_events::{EventsReceived, StreamClosedError},
+    sources::opentelemetry::LOGS,
+    SourceSender,
+};
+
+#[derive(Clone)]
+pub(super) struct Service {
+    pub pipeline: SourceSender,
+    pub acknowledgements: bool,
+    pub events_received: Registered<EventsReceived>,
+    pub log_namespace: LogNamespace,
```

Comment:
:thinkies: 

I'm browsing our usage of the visibility modifiers and I'm not really seeing cases where the struct members are defined with a broader scope than the struct itself... I'm curious the reason to not make the members `pub(super)` or just not pub at all if it's not necessary ? Are the places in the project where we are doing 


```Rust

pub(super) struct Foo {
   pub(super) bar: usize,
}
```

, unnecessary?

Reply:
I'm unsure - I believe that was pulled in from master, @bruceg's code for the `Registered<EventsReceived>` I think.